subsys/bluetooth/controller/ecdh.c

```diff
@@ -0,0 +1,338 @@
+#include <zephyr.h>
+#include <sys/atomic.h>
+
+#include <sys/byteorder.h>
+
+#include <bluetooth/hci.h>
+#include <bluetooth/buf.h>
+#include <bluetooth/hci_err.h>
+#include <drivers/bluetooth/hci_driver.h>
+
+#include <mpsl/mpsl_work.h>
+
+#include "ecdh.h"
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER)
+#define LOG_MODULE_NAME bt_ecdh
+#include "common/log.h"
+
+static struct {
+	uint8_t private_key_be[32];
+
+	union {
+		uint8_t public_key_be[64];
+		uint8_t dhkey_be[32];
+	};
+} ecdh;
+
+enum {
+	GEN_PUBLIC_KEY  = BIT(0),
```

Comment:
Why do these values have to be consecutive bits ?

Reply:
They had to before, but not anymore. I'll leave it as is for now.